### PR TITLE
Drag and Drop images into notebook markdown cells

### PR DESCRIPTION
**Pull request by Yoyokrazy**
Create and register DocumentDropEdit Provider, storing images as encoded base64 attachments. 

Currently the provider isn't being used at all, with the following settings being used:
```
    "ipynb.experimental.pasteImages.enabled": true,
    "editor.experimental.pasteActions.enabled": true,
    "editor.dropIntoEditor.enabled": true
```
cc @mattbierner, is there a missing bit of setup? Was looking at the drop example inside microsoft/vscode-extension-samples and couldn't find anything that I was missing on a first pass through.

### CODE REVIEW

File: extensions/ipynb/src/notebookImagePaste.ts [hunk 1]
```diff
@@ -66,19 +66,68 @@ class CopyPasteEditProvider implements vscode.DocumentPasteEditProvider {
...
+		if (!dataItem) {
+			return undefined;
 		}
-		return undefined;
+		const fileDataAsUint8 = await dataItem.asFile()?.data();
+		if (!fileDataAsUint8) {
+			return undefined;
+		}
+
+		// get filename data from paste
```
Comment (mjbvz) [inline extensions/ipynb/src/notebookImagePaste.ts:94]: A lot of this logic should be the same as paste. Can you extract this to a shared function both drop and paste can use

File: extensions/ipynb/src/notebookImagePaste.ts [hunk 2]
```diff
@@ -66,19 +66,68 @@ class CopyPasteEditProvider implements vscode.DocumentPasteEditProvider {
...
-				}
-			}
+class ImageDropEditProvider implements vscode.DocumentDropEditProvider {
+	async provideDocumentDropEdits(
+		document: vscode.TextDocument,
+		_position: vscode.Position,
+		dataTransfer: vscode.DataTransfer,
+		_token: vscode.CancellationToken
+	): Promise<vscode.DocumentDropEdit | undefined> {
+		const enabled = vscode.workspace.getConfiguration('ipynb', document).get('experimental.pasteImages.enabled', false);
```
Comment (mjbvz) [inline extensions/ipynb/src/notebookImagePaste.ts:78]: For the future, this likely should have its own setting (or the existing setting should have a more generic name)

**Review by mjbvz — CHANGES_REQUESTED** [review]
I suspect you are trying to drag from vscode's explorer. In that case, we don't create a `image/png` entry in the `DataTransfer`. Instead you get a `text/uri-list` of values

See how the markdown extension handles this:

https://github.com/microsoft/vscode/blob/8cf08fb35468dbc16a98d17bc01413d9376a6eb8/extensions/markdown-language-features/src/languageFeatures/dropIntoEditor.ts#L41

**Comment by Yoyokrazy** [comment]
This seems to be working, it required disabling `"markdown.editor.drop.enabled"`.  Relies on support from the cleaning tool as well, because it stores images in the same way. 

I imagine that there needs to be a bit more work done on dropping from `vsc file explorer -> cell` versus dropping from `windows file explorer -> cell`. This adds support for the latter, adding images from the vsc file explorer shouldn't use the `attachment:` syntax that pasted images do, as they will be stored locally and can be referenced via a path. Not sure how these two methods will interact with eachother, didn't have time to do thorough testing.

**Comment by Yoyokrazy** [comment]
image_drop

quick preview of drag n drop working

**Pull request closed without merging**